code/game/dna/dna2.dm

```diff
@@ -189,6 +190,40 @@ GLOBAL_LIST_EMPTY(bad_blocks)
 		val=rand(1, 2049)
 	SetUIValue(block, val, defer)
 
+//Get Tri State Block State
+/datum/dna/proc/GetUITriState(block)
+	if(block <= 0)
+		return
+	else
+		var/val = GetUIValue(block)
+		if(val >= 1395 && val <= 2760) //female
```

Comment:
Can make this a switch
```
switch(value)
  if(1 to 1395)
   ...
...
  else
   ...
```

Reply:
Not sure if I did this correctly, I'm going to DM you on Discord